### PR TITLE
Native Oculus Support for Core

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# XRTK - Mixed Reality Toolkit Change Request

## Overview

Native Oculus Support for Core

## Target of the change:

Is this enhancement for:

- Core (core framework, interfaces and definitions)

## Changes:

- New Oculus Submodule
- Package link to new submodule
- Added new extensions for InteractionMappings
- Updated InteractionMappings definitions to support custom named inputs

## Notes

Requires build pipeline setting up for Oculus repo prior to merge

### CODE REVIEW

**Comment by SimonDarksideJ** [comment]
Must not merge until the Oculus UPM package is registered and live.  Final step requires the UPM package registered with the Core XRTK for deployment.

File: .gitmodules [hunk 1]
```diff
@@ -11,3 +11,6 @@
 	path = Submodules/Lumin
 	url = https://github.com/XRTK/Lumin.git
 
+[submodule "Submodules/Oculus"]
```
Comment (StephenHodgson) [inline .gitmodules]: Why is there an extra line above?

File: XRTK-Core/Packages/com.xrtk.core/Definitions/Devices/MixedRealityInteractionMapping.cs [hunk 2]
```diff
@@ -193,6 +258,15 @@ public MixedRealityInputAction MixedRealityInputAction
         /// </summary>
         public KeyCode KeyCode => keyCode;
 
+        [SerializeField]
+        [Tooltip("Optional KeyCode value to get input from Unity's old input system.")]
+        private string inputName;
+
+        /// <summary>
+        /// Optional inputName value to get input for a coded input identity from a provider.
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Definitions/Devices/MixedRealityInteractionMapping.cs]: Shouldn't the summary match the tooltip description?

File: XRTK-Core/Packages/com.xrtk.core/Extensions/InteractionMappingsExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+using XRTK.Definitions.Devices;
+using XRTK.Definitions.Utilities;
+using XRTK.Interfaces.InputSystem;
+using XRTK.Services;
+
+namespace XRTK.Extensions
+{
+    /// <summary>
+    /// Extensions for the InteractionSource class to add haptics and expose the renderable model.
```
Comment (StephenHodgson) [inline XRTK-Core/Packages/com.xrtk.core/Extensions/InteractionMappingsExtensions.cs]: This summary doesn't match what's actually going on in this class

File: .gitmodules (hunk 1 above)
Comment (SimonDarksideJ) [reply .gitmodules]: Will fix once Oculus package resolved

File: XRTK-Core/Packages/com.xrtk.core/Definitions/Devices/MixedRealityInteractionMapping.cs (hunk 2 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Definitions/Devices/MixedRealityInteractionMapping.cs]: Will double check

File: XRTK-Core/Packages/com.xrtk.core/Extensions/InteractionMappingsExtensions.cs (hunk 3 above)
Comment (SimonDarksideJ) [reply XRTK-Core/Packages/com.xrtk.core/Extensions/InteractionMappingsExtensions.cs]: Doh, copy / paste error

**Comment by SimonDarksideJ** [comment]
Oculus repo now passing tests, Master/Release build submitted.  Once complete, this PR can move ahead and finally add Oculus support

**Comment by StephenHodgson** [comment]
@SimonDarksideJ build failed

**Comment by SimonDarksideJ** [comment]
Understood, needs updating post the UPM merge anyway, just testing

**Pull request merged**